src/displaycontroller.cpp

```diff
@@ -1324,4 +1368,106 @@ void IRAM_ATTR BitmappedDisplayController::absCopyToBitmap(int srcX, int srcY, B
 }
 
 
+void IRAM_ATTR BitmappedDisplayController::drawBitmapWithTransform(BitmapTransformedDrawingInfo const & drawingInfo, Rect & updateRect)
+{
+  // work out corners of the bitmap by taking the corners of the bitmap and transforming them by the matrix
+  int x = paintState().origin.X;
+  int y = paintState().origin.Y;
+  int width = drawingInfo.bitmap->width;
+  int height = drawingInfo.bitmap->height;
+
+  Rect originalBox = Rect(0, 0, width, height);
+  auto transformMatrix = drawingInfo.transformMatrix;
+
+  // work out each corner's new position
+  int minX = INT_MAX;
+  int minY = INT_MAX;
+  int maxX = INT_MIN;
+  int maxY = INT_MIN;
+  float pos[3] = { (float)originalBox.X1, (float)originalBox.Y1, 1.0f };
+  float transformed[3];
+  dspm_mult_f32(transformMatrix, pos, transformed, 3, 3, 1);
```

Comment:
all of the transforms in this function are done using the raw low-level esp-dsp matrix multiplication function.  this ensures we have control of the destination for the calculation.  using the `Mat` class is slightly simpler, but would be creating new objects for every calculation, and be a lot less efficient.

when using this form of the function the native esp32 machine code version should be used when running on the VDP - and a fall-back ANSI C implementation used inside the emulator